apps/s_server.c

```diff
@@ -3450,7 +3492,11 @@ static int rev_body(int s, int stype, int prot, unsigned char *context)
     BIO_push(io, ssl_bio);
     ssl_bio = NULL;
 #ifdef CHARSET_EBCDIC
-    io = BIO_push(BIO_new(BIO_f_ebcdic_filter()), io);
+    filter = BIO_new(BIO_f_ebcdic_filter());
+    if (filter == NULL)
+        goto err;
```

Comment:
I think you should add SSL_free(con) here as well.

Reply:
I don't think so.
The comment above says "No need to free |con| after this. Done by BIO_free(ssl_bio)".
Also the below just uses "goto err;" when fails.
Therefore, the 'con' and 'sbio' don't need to be freed manually.

Reply:
Ah, right. I've overlooked that.